templates/default/haproxy.cfg.erb

```diff
@@ -419,6 +419,13 @@ listen <%= key %>
 <% if listen['default_backend'] -%>
   default_backend <%= listen['default_backend'] %>
 <% end %>
+<% unless nil_or_empty?(listen['option']) %>
+<% listen['option'].each do | option |%>
+<% option.each do | option | %>
```

Comment:
I can't say I am running haproxy in my envs, the example seems to be a flat array but it looks like we are iterating as if its an array of arrays. Can this be an array of array options? If so could we add an example to the documentation?

Reply:
Got to agree, can we confirm this is correct as it looks like a mistake.

Reply:
Yeah it does look funny. I based it on [how it's done](https://github.com/sous-chefs/haproxy/blob/ccb6e9fe96688175b6e11eff8e2b3946c7daf8e3/templates/default/haproxy.cfg.erb#L297-L303) for `frontend` and `backend`. Looks like it's because `:option` takes an [Array](https://github.com/sous-chefs/haproxy/blob/ccb6e9fe96688175b6e11eff8e2b3946c7daf8e3/resources/backend.rb#L27) which then gets [pushed into another array](https://github.com/sous-chefs/haproxy/blob/ccb6e9fe96688175b6e11eff8e2b3946c7daf8e3/resources/backend.rb#L68). And so the template has to iterate over both array dimensions.

We have the same situation with `:acl` except it gets [flattened into a 1D array](https://github.com/sous-chefs/haproxy/blob/ccb6e9fe96688175b6e11eff8e2b3946c7daf8e3/templates/default/haproxy.cfg.erb#L335), hence no need for the extra iteration.

We could switch to the `flatten` approach for `option` or merge this as-is and do the cleanup later.

Reply:
Gotcha, I don't really have the time to dig into this to see why the code was written that way but it's certainly a way to do that. I lean towards using `flatten` as we have an established pattern and that is a little clearer as to the intent (without reading a lot of code) even if they are both equally correct/complete. That being said I could be talked into accepting it as is if other reviewers agree.